app/pages/projects/projects-search-selector.jsx

```diff
@@ -11,14 +11,12 @@ class SearchSelector extends Component {
     this.searchByName = this.searchByName.bind(this);
   }
 
-  navigateToProject(projectId) {
-    apiClient.type('projects').get(projectId)
-      .then(project => {
-        if (project.redirect != null && project.redirect.length !== 0) {
-          return browserHistory.push(project.redirect);
-        }
-        return browserHistory.push(['/projects', project.slug].join('/'));
-      });
+  navigateToProject(projectUrl) {
+    if (projectUrl.match(/^http.*/)) {
+      window.location.href = projectUrl;
```

Comment:
Because all you needed was the project slug / redirect that i pass in from the previous api calls as the value on the select (they are all uniq URL's / slugs).